File: leetcode/1143_longest_common_subsequence.py
# https://leetcode.com/problems/longest-common-subsequence/

class Solution:
    def longestCommonSubsequence(self, text1: str, text2: str) -> int:
        n1 = len(text1)
        n2 = len(text2)
        dp = [[0] * (n2 + 1) for _ in range(n1 + 1)]

        for i, t1 in enumerate(text1, 1):
            for j, t2 in enumerate(text2, 1):
                if t1 == t2:
                    dp[i][j] = 1 + dp[i-1][j-1]
                else:
                    dp[i][j] = max(dp[i-1][j], dp[i][j-1])

        return dp[n1][n2]

File: leetcode/1143_longest_common_subsequence.ts
// https://leetcode.com/problems/longest-common-subsequence/

function longestCommonSubsequence(text1: string, text2: string): number {
    let l1 = text1.length;
    let l2 = text2.length;
    let dp = new Array(l1 + 1).fill(0).map(() => new Array(l2 + 1).fill(0));

    for (const [i, t1] of ['', ...text1].entries()) {
        for (const [j, t2] of ['', ...text2].entries()) {
            if (i == 0 || j == 0) {
                continue;
            } else if (t1 == t2) {
                dp[i][j] = 1 + dp[i-1][j-1];
            } else {
                dp[i][j] = Math.max(dp[i-1][j], dp[i][j-1]);
            }
        }
    }
    return dp[l1+1][l2+1];
};
